src/gpuarray/buffer.h

```diff
@@ -757,6 +763,48 @@ typedef enum _ga_usefl {
   GA_USE_OPENCL =   0x4000,
 } ga_usefl;
 
+/************************************************************************************
+*                         multi-gpu communicator interface                         *
+************************************************************************************/
+
+/**
+ * \brief Creates a new communicator (multi process version)
+ * \param clique_id [const char*] TODO
+ * \param ndev [int] TODO
+ * \param rank [int] TODO
+ * \param res [int*] TODO
+ * \return gpucomm* TODO
+ */
+GPUARRAY_PUBLIC gpucomm* gpucomm_init(const char* clique_id, int ndev, int rank,
+                                      int* res);
```

Comment:
Maybe this should also take a `gpucontext *` to identify the device to work with.